### ISSUE TITLE
Constant segfaults after latest upgrade Ubuntu 14.04

### ISSUE THREAD

**Issue by r3wt** [facebook/hhvm#4460]
stack trace:

```
Host: development
ProcessID: 4095
ThreadID: 7f9282bff700
ThreadPID: 4098
Name: unknown program
Type: Segmentation fault
Runtime: hhvm
Version: tags/HHVM-3.4.1-0-g5b66f110ebe9bafccecd8946d391b0f895dc107a
DebuggerCount: 0

ThreadType: Web Request
Server_SERVER_NAME: bonieamaral.testebeta.com.br
Server: bonieamaral.testebeta.com.br
URL: /json/uc_get_type/restraunt/0

# 0  HPHP::Variant::assign(HPHP::Variant const&) at /usr/bin/hhvm:0
# 1  HPHP::MySQLStmt::fetch() at /usr/bin/hhvm:0
# 2  HPHP::c_mysqli_stmt_ni_fetch(HPHP::ObjectData*) at ext_mysqli.cpp:0
# 3  void HPHP::Native::callFunc<false, false>(HPHP::Func const*, void*, HPHP::TypedValue*, HPHP::TypedValue&) at /usr/bin/hhvm:0
# 4  HPHP::TypedValue* HPHP::Native::methodWrapper<false, false>(HPHP::ActRec*) at /usr/bin/hhvm:0
# 5  void HPHP::ExecutionContext::dispatchImpl<false>() at /usr/bin/hhvm:0
# 6  HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, HPHP::Func const*, HPHP::Variant const&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags) at /usr/bin/hhvm:0
# 7  HPHP::vm_call_user_func(HPHP::Variant const&, HPHP::Variant const&, bool) at /usr/bin/hhvm:0
# 8  HPHP::f_call_user_func_array(HPHP::Variant const&, HPHP::Variant const&) at /usr/bin/hhvm:0
# 9  HPHP::fg_call_user_func_array(HPHP::ActRec*) at /usr/bin/hhvm:0
# 10 void HPHP::ExecutionContext::dispatchImpl<false>() at /usr/bin/hhvm:0
# 11 HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, HPHP::Func const*, HPHP::Variant const&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags) at /usr/bin/hhvm:0
# 12 HPHP::ExecutionContext::invokeUnit(HPHP::TypedValue*, HPHP::Unit const*) at /usr/bin/hhvm:0
# 13 HPHP::include_impl_invoke(HPHP::String const&, bool, char const*) at /usr/bin/hhvm:0
# 14 HPHP::hphp_invoke(HPHP::ExecutionContext*, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, HPHP::Array const&, HPHP::VRefParamValue const&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool&, std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool) at /usr/bin/hhvm:0
# 15 HPHP::HttpRequestHandler::executePHPRequest(HPHP::Transport*, HPHP::RequestURI&, HPHP::SourceRootInfo&, bool) at /usr/bin/hhvm:0
# 16 HPHP::HttpRequestHandler::handleRequest(HPHP::Transport*) at /usr/bin/hhvm:0
# 17 HPHP::ServerWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::FastCGITransportTraits>::doJob(std::shared_ptr<HPHP::FastCGIJob>) at /usr/bin/hhvm:0
# 18 HPHP::JobQueueWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::Server*, true, false, HPHP::JobQueueDropVMStack>::start() at /usr/bin/hhvm:0
# 19 HPHP::AsyncFuncImpl::ThreadFunc(void*) at /usr/bin/hhvm:0
# 20 HPHP::start_routine_wrapper(void*) at /usr/bin/hhvm:0
# 21 start_thread at /lib/x86_64-linux-gnu/libpthread.so.0:0
# 22 clone at /lib/x86_64-linux-gnu/libc.so.6:0
```

Code worked fine under previous 3.3.0 package

**Comment by mikelbring** [comment]
@r3wt Did you determine using a socket worked a lot better? I am having this issue frequently.

**Comment by r3wt** [comment]
@mikelbring i never was able to fix it. the app i was working on ended development cycle, and i guess they just decided to ship it on the 3.3 branch of HHVM, with PHP5 as failover

**Comment by jwatzman** [comment]
@mikelbring -- still interested in more info on this if you're hitting it. See my comment https://github.com/facebook/hhvm/issues/4460#issuecomment-67210807 for what we need to have a chance to look into this.

**Comment by paulbiss** [comment]
This looks like #4505, it should be fixed in master.

**Comment by jwatzman** [comment]
Closing as fixed by #4505.
